mindsdb/integrations/utilities/rag/chains/map_reduce_summarizer_chain.py

```diff
@@ -154,11 +178,17 @@ def _call(
         source_id_to_summary = {}
         for summary in summaries:
             source_id_to_summary[summary.source_id] = summary.content
+        logger.debug(f"Generated {len(source_id_to_summary)} summaries")
 
         # Update context chunks with document summaries.
         for chunk in context_chunks:
-            doc_id = chunk.metadata.get(self.doc_id_key, '')
-            summary = source_id_to_summary[str(doc_id)]
-            chunk.metadata['summary'] = summary
+            metadata = chunk.metadata or {}
```

Comment:
This line looks a bit weird, is this a standard way to check for None?

Reply:
modified it to be more explicit